src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoSlickImpl/FileIO.scala

```diff
@@ -0,0 +1,122 @@
+package de.htwg.se.sudoku.model.fileIoComponent.fileIoSlickImpl
+
+import com.google.inject.{Guice, Inject}
+import com.google.inject.name.{Named, Names}
+import de.htwg.se.sudoku.{SlickModule}
+import de.htwg.se.sudoku.model.database.{Cell, Grid}
+import de.htwg.se.sudoku.model.fileIoComponent.FileIOInterface
+import de.htwg.se.sudoku.model.gridComponent.GridInterface
+import net.codingwell.scalaguice.InjectorExtensions.ScalaInjector
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+import slick.jdbc.H2Profile.api._
+import slick.jdbc.meta.MTable
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class FileIO @Inject()(@Named("H2Url") url: String, @Named("H2User") dbUser: String) extends FileIOInterface{
+
+  val db =  Database.forURL(url, user = dbUser)
+
+  // create schema if it doesn't exist
+  val tables = List(
+    TableQuery[Grid],
+    TableQuery[Cell]
+  )
+
+  val existingTables = db.run(MTable.getTables)
```

Comment:
`existingTables` and `createSchemaFuture` needs in my IDE type annotations because they are public members...